### ISSUE TITLE
`redundant_pattern_matching` applies redundant parenthesis to expression with `?`

### ISSUE THREAD

**Issue by mks-h** [rust-lang/rust-clippy#16045]
### Summary

There's no point to wrapping `expression?` in parenthesis, since `(expression?).call()` and `expression?.call()` work the same, and the latter is... Well, simply better.

### Lint Name

redundant_pattern_matching

### Reproducer

I tried this code:

```rust
use std::io;

struct FailableIterator {}

impl FailableIterator {
    fn next_result(&mut self) -> Result<Option<usize>, io::Error> {
        Ok(None)
    }
}

fn main() -> Result<(), io::Error> {
    let mut fiter = FailableIterator {};
    while let Some(_) = fiter.next_result()? {}
    Ok(())
}
```

I saw this happen:

```
    Checking playground v0.0.1 (/playground)
warning: redundant pattern matching, consider using `is_some()`
  --> src/main.rs:13:15
   |
13 |     while let Some(_) = fiter.next_result()? {}
   |     ----------^^^^^^^----------------------- help: try: `while (fiter.next_result()?).is_some()`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.91.0/index.html#redundant_pattern_matching
   = note: `#[warn(clippy::redundant_pattern_matching)]` on by default
```

I expected to see this happen:

```
    Checking playground v0.0.1 (/playground)
warning: redundant pattern matching, consider using `is_some()`
  --> src/main.rs:13:15
   |
13 |     while let Some(_) = fiter.next_result()? {}
   |     ----------^^^^^^^----------------------- help: try: `while fiter.next_result()?.is_some()`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.91.0/index.html#redundant_pattern_matching
   = note: `#[warn(clippy::redundant_pattern_matching)]` on by default
```

### Version

```text
https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=2b291bbdcdecba2e6fd1f2dd16434242
```

**Comment by mks-h** [comment]
Oops, set a wrong tag. Cannot change them after filing, sadly.

**Comment by ada4a** [comment]
I'm assuming what you wanted to add is `C-enhancement`? If so, you can do that as follows:

@rustbot label -I-false-positive +C-enhancement

See https://forge.rust-lang.org/triagebot/labeling.html for more information

**Comment by ada4a** [comment]
Also, the playground link doesn't show the Rust version used unfortunately -- but you _can_ get it by adding this:
```rs
fn main() {
    let _ = std::process::Command::new("rustc").arg("-vV").spawn();
}
```
and hitting "Run"
